modules/Emails/EmailsController.php

```diff
@@ -636,11 +636,14 @@ public function action_DisplayDetailView()
     {
         $db = DBManagerFactory::getInstance();
         $emails = BeanFactory::getBean("Emails");
-        
-        $inboundEmailRecordIdQuoted = $db->quote($_REQUEST['inbound_email_record']);
-        $uidQuoted = $db->quote($_REQUEST['uid']);
-        
-        $result = $emails->get_full_list('', "mailbox_id = '" . $inboundEmailRecordIdQuoted . "' AND uid = '" . $uidQuoted . "'");
+
+        $inboundEmailRecordId = $_REQUEST['inbound_email_record'];
+
+        $uid = $_REQUEST['uid'];
+
+        $subQuery = "`mailbox_id` = " . $db->quoted($inboundEmailRecordId) . " AND `uid` = " . $db->quoted($uid);
```

Comment:
I'm sorry I couldn't notice it earlier but I just want to let you know you can use `SuiteValidator::isValidId()` for any suitecrm id validation before using it. This validation result also implicate that if you can using it as a valid ID in a query before requesting the sql server to resolve the queries in these areas (invalid ID obiously will failing so we don't have to waiting for the mysql server responses). It could improve the performance too.